File: app/Http/Controllers/SpecialistPatterns/LR/LR1SP/graph/ColorGOFactory.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\LR\LR1SP\graph;

// Класс ColorGOFactory представляет собой фабрику для создания цветных графических объектов.
class ColorGOFactory extends AbstractGOFactory
{
    // Приватное статическое свойство для хранения значения по умолчанию для точки.
    private static ?Point $DEFAULT_POINT = null;

    // Конструктор класса. Устанавливает значение по умолчанию для точки.
    public function __construct() {
        self::$DEFAULT_POINT = new Point(10, 20);
    }

    // Метод для создания объекта класса Point.
    public function createPoint(): Point
    {
        // Клонируем точку из значения по умолчанию и добавляем ее в сцену.
        $p = self::$DEFAULT_POINT->clone();
        Scene::getInstance()->add($p);
        return $p;
    }

    // Метод для создания объекта класса Circle.
    public function createCircle(): Circle
    {
        // Создаем круг с начальными параметрами и добавляем его в сцену.
        $c = new Circle(0, 0, 1);
        Scene::getInstance()->add($c);
        return $c;
    }
}

File: app/Http/Controllers/SpecialistPatternsJava/Lab1/Lab1Java/src/graph/ColorGOFactory.java
package graph;

// Класс ColorGOFactory представляет собой фабрику для создания цветных графических объектов.
public class ColorGOFactory extends AbstractGOFactory {
    // Статическое поле для хранения объекта Point по умолчанию.
    public static final Point DEFAULT_POINT = new Point(0, 0);

    // Переопределение метода для создания объекта Point.
    @Override
    public Point createPoint() {
        Point p = DEFAULT_POINT.clone(); // Клонируем объект Point по умолчанию.
        Scene.instance.add(p); // Добавляем точку в сцену.
        return p;
    }

    // Переопределение метода для создания объекта Circle.
    @Override
    public Circle createCircle() {
        Circle c = new Circle(0, 0, 1); // Создаем окружность с начальными параметрами.
        Scene.instance.add(c); // Добавляем окружность в сцену.
        return c;
    }
}
